nomenclature/testing.py

```diff
@@ -52,15 +52,26 @@ def assert_valid_structure(
     Folder structure of `path`:
         - A `definitions` folder is required and must be a valid
         :class:`DataStructureDefinition`
+        - The `definitions` folder must contain sub-folder(s) to validate
         - If a `mappings` folder exists, it must be a valid :class:`RegionProcessor`
 
     """
     if not (path / definitions).is_dir():
         raise NotADirectoryError(
             f"Definitions directory not found: {path / definitions}"
         )
-    if dimensions is None:
+
+    if dimensions is None:  # if "dimensions" were not specified
         dimensions = [x.stem for x in (path / definitions).iterdir() if x.is_dir()]
+        if not dimensions:
+            raise FileNotFoundError(
+                f"`definitions` directory is empty: {path / definitions}"
+            )
+
+    if dimensions == []:  # if "dimensions" were specified as "[]"
```

Comment:
I think that it would be more intuitive to do this check first (if user provided an empty list, raise an error; then continue to see whether there are folders to validate...)